fwoa-tools/src/migrationVerify.ts

```diff
@@ -0,0 +1,132 @@
+const argv: any = parseCmdOptions();
+const smartClient: boolean = argv.smart;
+const dryRun: boolean = argv.dryRun;
+const logs: string[] = [];
+
+async function verifyFolderImport(): Promise<void> {
+  const outputFile: ExportOutput = JSON.parse(readFileSync('migrationExport_Output.txt').toString());
+  const fileNames = outputFile.file_names;
+
+  Object.entries(fileNames).forEach(async ([resourceType, resourcePaths]) => {
+    console.log(`Starting import for resource ${resourceType}`);
+    const interceptor = aws4Interceptor({
+      region: API_AWS_REGION!,
+      service: 'healthlake'
+    });
+
+    const fhirClient = await (smartClient ? getFhirClientSMART() : getFhirClient());
+    const s3Client = new S3({
+      region: API_AWS_REGION!
+    });
+    const healthLakeClient = axios.create();
+    healthLakeClient.interceptors.request.use(interceptor);
+
+    // Retrieve resource from HealthLake and compare it to fwoa.
+    let i: number = 0;
+    for (i; i < resourcePaths.length; i += 1) {
+      // resource path includes jobId, tenantId, resourceType, and S3 object id
+      // eslint-disable-next-line security/detect-object-injection
+      const resourcePath = resourcePaths[i];
+      logs.push(`${new Date().toISOString()}: Verifying Import from ${resourcePath}...`);
+      const resourceFile = await s3Client
+        .getObject({
+          Bucket: IMPORT_OUTPUT_S3_BUCKET_NAME!,
+          Key: resourcePath
+        })
+        .promise();
+      if (resourceFile.$response.error) {
+        throw new Error(`Failed to read file ${resourceFile.$response.error}`);
+      }
+
+      // Each resource file can contain a number of resource objects
+      const allResources: string[] = resourceFile.Body!.toString().split('\n');
+      let j: number = 0;
+      for (j; j < allResources.length; j += 1) {
+        // eslint-disable-next-line security/detect-object-injection
+        const resource = JSON.parse(allResources[j]);
+        const id = resource.id;
+        const resourceInHL = await healthLakeClient.get(`${DATASTORE_ENDPOINT}/${resourceType}/${id}`);
+        logs.push(
+          `${new Date().toISOString()}: Retrieved resource at ${resourcePath} from datastore, comparing to FWoA...`
+        );
+        if (!(await verifyResource(fhirClient, resourceInHL.data, id, resourceType))) {
+          throw new Error(`Resources in FWoA and AHL do not match, ${resourcePath}`);
+        }
+      }
+    }
+  });
+}
+
+async function checkConfiguration(): Promise<void> {
+  if (!DATASTORE_ID) throw new Error('DATASTORE_ID environment variable is not defined');
+  if (!DATASTORE_ENDPOINT) throw new Error('DATASTORE_ENDPOINT environment variable is not defined');
+  if (!API_AWS_REGION) throw new Error('API_AWS_REGION environment variable is not defined');
+  if (!IMPORT_OUTPUT_S3_BUCKET_NAME)
+    throw new Error('IMPORT_OUTPUT_S3_BUCKET_NAME environment variable is not defined');
+
+  const healthLake: HealthLake = new HealthLake({
```

Comment:
Could we also verify the FHIR client as well for redundancy here?

Reply:
added!